src/weather.cpp

```diff
@@ -36,7 +37,8 @@ const efftype_id effect_glare( "glare" );
 void weather_effect::glare()
 {
     if( PLAYER_OUTSIDE && g->is_in_sunlight( g->u.pos() ) && !g->u.in_sleep_state() &&
-        !g->u.worn_with_flag( "SUN_GLASSES" ) && !g->u.has_bionic( "bio_sunglasses" ) ) {
+        !g->u.worn_with_flag( "SUN_GLASSES" ) && !g->u.worn_with_flag( "BLIND" ) &&
+        !g->u.has_bionic( "bio_sunglasses" ) && !g->u.has_effect( effect_blind ) ) {
```

Comment:
The blindness check is repeated multiple times in the code, it would be much better if it was in a member function of `Creature` class (since `monster` already has `is_blind`).